### PR TITLE
Root cause System.Net.Http test failures on NETFX

### LINKED ISSUE #17691: Desktop and Core behavior differences in System.Net.Http
There are some differences in behavior between Desktop and core which are causing tests like this one fail on Desktop:

https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs#L281

cc: @davidsh

### PR DESCRIPTION
**Pull request by davidsh**
Finished investigation of failures on .NET Framework for
System.Net.Http.

Added specific NETFX test changes where necessary. Opened up additional
bugs for investigation.

Added missing parameter validation for HttpClientHandler.MaxConnectionsPerServer
for the net46 OOB. This matches .NET Core now.

Removed some test code that was incorrectly trying to do CredentialCache.Add()
with a 'basic' authtype and using DefaultCredentials. This is invalid and
throws exception on netfx. Added #18870.

Remove PlatformNotSupportedProxy test code. It was throwing an exception
on netfx and the purpose of it didn't seem to make sense in the one test
that was using it.

Fixes #17691.

### CODE REVIEW

**Comment by davidsh** [comment]
@dotnet-bot test outerloop netcoreapp Windows_NT Debug
@dotnet-bot test outerloop netcoreapp Ubuntu16.04 Debug

**Comment by davidsh** [comment]
@safern @danmosemsft @stephentoub PTAL

File: src/System.Net.Http/tests/FunctionalTests/CancellationTest.cs [hunk 1]
```diff
@@ -61,13 +60,27 @@ public CancellationTest(ITestOutputHelper output)
...
                     var stopwatch = Stopwatch.StartNew();
-                    await Assert.ThrowsAnyAsync<OperationCanceledException>(async () =>
+                    if (PlatformDetection.IsFullFramework)
                     {
-                        Task<HttpResponseMessage> getResponse = client.GetAsync(url, HttpCompletionOption.ResponseContentRead, cancellationToken);
-                        await triggerRequestCancel.Task;
-                        cts.Cancel();
-                        await getResponse;
-                    });
+                        // .NET Framework throws WebException instead of OperationCancledException.
```
Comment (danmosemsft) [inline src/System.Net.Http/tests/FunctionalTests/CancellationTest.cs]: utter nit, typo

Comment (davidsh) [inline src/System.Net.Http/tests/FunctionalTests/CancellationTest.cs]: fixed.

**Comment by davidsh** [comment]
@dotnet-bot test outerloop netcoreapp Windows_NT Debug
@dotnet-bot test outerloop netcoreapp Ubuntu16.04 Debug

**Pull request merged**